src/client/include/com/amazonaws/kinesis/video/client/Include.h

```diff
@@ -2061,6 +2095,30 @@ PUBLIC_API STATUS getKinesisVideoStreamData(STREAM_HANDLE, UPLOAD_HANDLE, PBYTE,
  */
 PUBLIC_API STATUS putKinesisVideoFragmentMetadata(STREAM_HANDLE, PCHAR, PCHAR, BOOL);
 
+/**
+ * Inserts a "metadata" - a key/value string pair into the stream.
+ *
+ * NOTE: The metadata are modelled as MKV tags and are not immediately put into the stream as
+ * it might break the fragment.
+ * This is a limitation of MKV format as Tags are level 1 elements.
+ * Instead, they will be accumulated and inserted in-between the fragments and at the end of the stream.
+ *
+ * MKV spec is available at: https://matroska.org/technical/specs/index.html
+ *
+ * Putting a "persistent" metadata will result in the metadata being inserted before every fragment.
+ * The metadata can be changed by calling this function with the same name and a different value.
+ * Specifying an empty string for the value for a persistent metadata will clear it and it won't
+ * be applied to the consecutive fragments.
+ *
+ * @param 1 STREAM_HANDLE - the stream handle.
+ * @param 2 PCHAR - the metadata name.
+ * @param 3 PCHAR - the metadata value.
+ * @param 4 BOOL - Whether to keep applying the metadata to following fragments.
+ *
+ * @return Status of the function call.
+ */

```

Comment:
The comments here are copy pasted from existing method but needs to be updated, the params are different.